### ISSUE TITLE
Random variable arithmetic and Cholesky factors

### ISSUE THREAD

**Issue by pnkraemer** [probabilistic-numerics/probnum#335]
TL;DR: This is a more thorough version of #319. 

With #331, it is easier to check from the outside whether a Normal RV has a Cholesky factor or not. 
Arithmetic involving `Normal` RVs currently ignores this factor (see e.g. #319), which leads to a loss of information whenever computation relies on Cholesky factors (some do quite heavily, e.g. the square-root Kalman filter). Little by little, I think we should change this. To this end, I suppose we need to do
- [ ] Move `cholesky_update` out of `probnum.statespace` into a more basic module, so it can be used more widely (`probnum.utils`?). At some point, maybe we move it to `marvinpfoertner/cholupdates` (see https://github.com/marvinpfoertner/cholupdates/issues/14) but this is a different issue. 
- [ ] Implement an updated version of **constant-normal arithmetic** (addition and multiplication) that uses Cholesky factors
- [ ] Implement an updated version of **addition of normals** that uses Cholesky factors
- [ ] Implement an updated version of **slicing of multivariate normals**, that uses Cholesky factors
- [ ] Implement an updated version of **reshaping and transposing of normals** that uses Cholesky factors

I'll try to get started with the first one or two points in the near future. I do not know whether I will have time for all of these points, so some help may be needed. 

Also, as soon as it comes to e.g. addition of normals, there are decisions to make (e.g., what should happen if one of the RVs has a Cholesky factor and the other one does not).
